### ISSUE TITLE
With WebView2 inside a C# DialogForm : after call to Hide(),  WebView2 stay visible

### ISSUE THREAD

**Issue by Phildre92** [MicrosoftEdge/WebView2Feedback#828]
**Description**
I have a WebView2 inside a  C# DialogForm.
The Hide() Function close the  DialogForm but WebView2 window stays visible as a top level window.
There a similar problem with Show() for a DialogForm, WebView2 stays invisible.

To "solve" this problem, I have managed myself visibility of WebView2 window :
- using DialogForm.Shown Callback to add an explicit call to WebView2.Show()
- using DialogForm.FormClosed to add an explicit call to WebView2.Hide

There is also a similar problem when the WebView2 is in TabPanel.

**Version**
Microsoft Visual Studio Community 2019
Version 16.8.2
VisualStudio.16.Release/16.8.2+30717.126
Microsoft .NET Framework
Version 4.8.04084
Visual C++ 2019   00435-60000-00000-AA288
Microsoft Visual C++ 2019
Package Microsoft Visual Studio VC   1.0
Package Microsoft Visual Studio VC

SDK: Microsoft.Web.WebView2 1.0.674-prerelease

Runtime: Canary Version 89.0.767.0 (Version officielle)  (64 bits)

Framework: WinForms

OS: Win10

[AB#31680887](https://microsoft.visualstudio.com/90b2a23c-cab8-4e7c-90e7-a977f32c1f5d/_workitems/edit/31680887)

**Comment by Alex240** [comment]
I had the same problem (VB.NET), the bug seems random.

**Comment by oggy22** [comment]
Thank you for reporting this issue. Having our WebView2 inside a ShowDialog does seem as something we didn't quite cover with our testing. We will work on it and get back to you.

**Comment by SamuelQZQ** [comment]
@Phildre92 What's the namespace of the DialogForm you mentioned? Is that Microsoft.Web.Management.Client.Win32.DialogForm? I'm trying to repo the bug. But I haven't found this control.

**Comment by Phildre92** [comment]
I'm using "Microsoft.Web.WebView2.WinForms.WebView2 " in a "System.Windows.Forms".
The "System.Windows.Forms" is managed using "ShowDialog()" to be modal.
To open/close this "WebView2" included this modal form, i need to add callbacks "System.EventHandler(this.Item_Form_Shown)" to the form included call to "show()" on "WebView2"  and "System.Windows.Forms.FormClosedEventHandler(this.Item_Form_FormClosed)" calling  "Hide()" on "WebView2".
"WebView2" is shown on the first call to "ShowDialog()", but then not hidden when closing the form clicking on the window manager cross  in the upper right corner of the form window.
Adding manual "Hide()" in closed callback of the form hides "WebView2" but then  "ShowDialog()"  sowh the form but not "WebView2".

**Comment by iamshz97** [comment]
Same Issue w/ ME

```CSharp
         using (var myWebView2= new myWebView2())
                {
                    carfaxForm.ShowDialog();
                }
```

Helped my case.

**Comment by iamshz97** [comment]
Same Issue w/ ME

         using (var myWebView2 = new myWebView2Form())
                {
                    myWebView2 .ShowDialog();
                }
Helped my case.

**Comment by iamshz97** [comment]
Same Issue w/ ME
```CSharp
         using (var myWebView2 = new myWebView2())
                {
                    carfaxForm.ShowDialog();
                }
```
Helped my case.

**Comment by styletronix** [comment]
It's also when the Form is displayed as MDI Child in a tabbed MDI Controller. When there is no other Window inside the tabbed MDI Container the WebView is hidden when form.hide() is called, but if there is another form in the tabbed mdi container, the webview stays on top as a seperate formless window. It reattaches itself if the containig form is shown again.

Setting form.mdiparent = nothing before hiding the form and reattaching it to the mdiparent when shown solved the issue, but that is only a workarround.

**Comment by champnic** [comment]
This should be fixed in 1.0.902-prerelease SDK. Thanks!
